### PR TITLE
*: support revoke statement

### PR DESCRIPTION
**Pull request by tiancaiamao**
Support revoke statement, the opposite to grant statement.

### CODE REVIEW

File: executor/grant.go [hunk 1]
```diff
@@ -123,44 +123,44 @@ func (e *GrantExec) Close() error {
...
 	// Entry does not exist for user-host-db. Insert a new entry.
-	return initDBPrivEntry(e.ctx, user, host, db.Name.O)
+	return initDBPrivEntry(ctx, user, host, db.Name.O)
 }
 
 // Check if table scope privilege entry exists in mysql.Tables_priv.
 // If unexists, insert a new one.
-func (e *GrantExec) checkAndInitTablePriv(user string, host string) error {
-	db, tbl, err := e.getTargetSchemaAndTable()
+func checkAndInitTablePriv(ctx context.Context, dbName, tableName string, is infoschema.InfoSchema, user string, host string) error {
```
Comment (hanfei1991) [inline executor/grant.go]: Can we let `GrantExec` and `RevokeExec` compose same structure to avoid change like this?

Comment (tiancaiamao) [inline executor/grant.go]: There many repeat code now, I've thought this problem.
But I found that `GrantExec` and `RevokeExec` behavior differently, I decide not to merge them to reuse code.

File: executor/revoke.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+func (e *RevokeExec) revokeOneUser(user, host string) error {
+	// If there is no privilege entry in corresponding table, insert a new one.
+	// DB scope:		mysql.DB
+	// Table scope:		mysql.Tables_priv
+	// Column scope:	mysql.Columns_priv
+	switch e.Level.Level {
+	case ast.GrantLevelDB:
+		err := checkAndInitDBPriv(e.ctx, e.Level.DBName, e.is, user, host)
```
Comment (coocood) [inline executor/revoke.go]: Revoke should not InitDBPriv.

**Comment by tiancaiamao** [comment]
PTAL @coocood

**Comment by shenli** [comment]
Please update statement metrics.

**Comment by shenli** [comment]
LGTM

**Pull request merged**